### PR TITLE
Trend line list updates

### PR DESCRIPTION
**Pull request by spireken**
I added a ChartingUiTester-TrendList demo.

### CODE REVIEW

File: Applications/Spire/Include/Spire/Ui/DropDownItem.hpp [hunk 1]
```diff
@@ -38,24 +38,26 @@ namespace Spire {
       */
       DropDownItem(QVariant value, QImage icon, QWidget* parent = nullptr);
 
+      virtual ~DropDownItem() = default;
```
Comment (kranar) [inline Applications/Spire/Include/Spire/Ui/DropDownItem.hpp:41]: Remove this.

File: Applications/Spire/Include/Spire/Ui/StaticDropDownMenu.hpp [hunk 2]
```diff
@@ -41,6 +41,8 @@ namespace Spire {
       StaticDropDownMenu(std::vector<QVariant> items,
         const QString& display_text, QWidget* parent = nullptr);
 
+      virtual ~StaticDropDownMenu() = default;
```
Comment (kranar) [inline Applications/Spire/Include/Spire/Ui/StaticDropDownMenu.hpp:44]: Why was this added?

**Review by kranar — CHANGES_REQUESTED** [review]

File: Applications/Spire/Include/Spire/Ui/StaticDropDownMenu.hpp (hunk 2 above)
Comment (spireken) [reply Applications/Spire/Include/Spire/Ui/StaticDropDownMenu.hpp:44]: Right, forgot about the base class' virtual destructor.

File: Applications/Spire/Include/Spire/Ui/DropDownItem.hpp [hunk 3]
```diff
@@ -39,23 +39,23 @@ namespace Spire {
...
+      virtual void reset_highlight();
 
       //! Connects a slot to the highlighted signal.
-      boost::signals2::connection connect_highlighted_signal(
+      virtual boost::signals2::connection connect_highlighted_signal(
         const HighlightedSignal::slot_type& slot) const;
       
       //! Connects a slot to the selected signal.
-      boost::signals2::connection connect_selected_signal(
+      virtual boost::signals2::connection connect_selected_signal(
```
Comment (kranar) [inline Applications/Spire/Include/Spire/Ui/DropDownItem.hpp:58]: Why are these now virtual? In general I do not like to see changes to an API that have not been discussed/approved.

File: Applications/Spire/Source/Charting/StyleDropDownMenuItem.cpp [hunk 4]
```diff
@@ -1,69 +1,40 @@
...
+StyleDropDownMenuItem::StyleDropDownMenuItem(TrendLineStyle style,
+  QWidget* parent)
+  : DropDownItem(QVariant::fromValue(style), parent),
+    m_style(style) {}
 
-void StyleDropDownMenuItem::mouseReleaseEvent(QMouseEvent* event) {
-  if(event->button() == Qt::LeftButton) {
-    m_selected_signal(m_style);
-  }
+// TODO: still required?
```
Comment (kranar) [inline Applications/Spire/Source/Charting/StyleDropDownMenuItem.cpp:25]: Resolve this TODO

**Review by kranar — CHANGES_REQUESTED** [review]

File: Applications/Spire/Include/Spire/Ui/DropDownItem.hpp (hunk 3 above)
Comment (spireken) [reply Applications/Spire/Include/Spire/Ui/DropDownItem.hpp:58]: I subclass the DropDownItem for items that require custom widgets or painting. Doesn't that require the base class to have virtual methods when used polymorphically?

Comment (spireken) [reply Applications/Spire/Include/Spire/Ui/DropDownItem.hpp:58]: I would argue they should have been virtual to begin with, but the other widget that subclasses the DropDownItem, SecurityInfoItem, doesn't override any of the base class methods (aside from the protected Qt event functions).

Comment (kranar) [reply Applications/Spire/Include/Spire/Ui/DropDownItem.hpp:58]: Don't need these to be ```virtual```.

Comment (spireken) [reply Applications/Spire/Include/Spire/Ui/DropDownItem.hpp:58]: Why?

Comment (kranar) [reply Applications/Spire/Include/Spire/Ui/DropDownItem.hpp:58]: The default is non-```virtual```, if method is declared to be ```virtual``` there needs to be a justification but I don't see one.

Comment (spireken) [reply Applications/Spire/Include/Spire/Ui/DropDownItem.hpp:58]: So is the standard we declare functions as virtual on an as-needed basis?

Comment (kranar) [reply Applications/Spire/Include/Spire/Ui/DropDownItem.hpp:58]: Yep! That's exactly it.

**Review by kranar — APPROVED** [review]
